pkg/cmd/inspect/inspect.go

```diff
@@ -167,102 +167,138 @@ func (o *InspectOptions) Run() error {
-		}
-		if info.Mapping.Resource.Resource != "clusteroperators" {
-			return fmt.Errorf("unsupported resource type, must be %q", "clusteroperators")
-		}
-	}
-
-	// next, ensure we're able to proceed writing data to specified destination
+	// ensure we're able to proceed writing data to specified destination
 	if err := o.ensureDirectoryViable(o.baseDir, o.overwrite); err != nil {
 		return err
 	}
 
-	// gather config.openshift.io resource data
-	skippedNamespaces := []string{}
+	// first, gather config.openshift.io resource data
 	errs := []error{}
 	if err := o.gatherConfigResourceData(path.Join(o.baseDir, "/cluster-scoped-resources/config.openshift.io")); err != nil {
 		errs = append(errs, err)
 	}
 
-	// gather operator.openshift.io resource data
+	// then, gather operator.openshift.io resource data
 	if err := o.gatherOperatorResourceData(path.Join(o.baseDir, "/cluster-scoped-resources/operator.openshift.io")); err != nil {
 		errs = append(errs, err)
 	}
 
+	// finally, gather polymorphic resources specified by the user
+	allErrs := []error{}
+	notFoundNamespaces := []string{}
 	for _, info := range infos {
-		// save clusteroperator resources
-		if err := o.gatherClusterOperatorResource(path.Join(o.baseDir, "/clusteroperator"), info); err != nil {
-			errs = append(errs, err)
-			continue
+		skippedNs, err := InspectResource(info, NewResourceContext(), o)
+		notFoundNamespaces = append(notFoundNamespaces, skippedNs...)
+		if err != nil {
+			allErrs = append(allErrs, err)
 		}
+	}
 
-		namespaces, err := obtainClusterOperatorNamespaces(info.Object)
-		if err != nil {
-			errs = append(errs, err)
-			continue
+	if len(notFoundNamespaces) > 0 {
+		for _, namespace := range notFoundNamespaces {
+			log.Printf("Data collection skipped namespace %q. Unable to find namespace...\n", namespace)
+		}
+	}
+	if len(allErrs) > 0 {
+		return errors.NewAggregate(allErrs)
+	}
+
+	log.Printf("Finished successfully with no errors.\n")
+	return nil
+}
+
+// InspectResource receives an object to gather debugging data for, and a context to keep track of
+// already-seen objects when following related-object reference chains.
+func InspectResource(info *resource.Info, context *resourceContext, o *InspectOptions) ([]string, error) {
+	if context.visited.Has(infoToContextKey(info)) {
+		return nil, nil
+	}
+	context.visited.Insert(infoToContextKey(info))
+
+	unstr, ok := info.Object.(*unstructured.Unstructured)
+	if !ok {
+		return nil, fmt.Errorf("unexpected type. Expecting %q but got %T", "*unstructured.Unstructured", info.Object)
+	}
+
+	notFoundNamespaces := []string{}
+
+	switch info.ResourceMapping().Resource.GroupResource() {
+	case configv1.GroupVersion.WithResource("clusteroperators").GroupResource():
+		// save clusteroperator resources to disk
+		if err := o.gatherClusterOperatorResource(path.Join(o.baseDir, "/cluster-scoped-resources", "/"+unstr.GroupVersionKind().Group, "/clusteroperators"), unstr); err != nil {
```

Comment:
make this idiot proof.  Only pass the basedir, the rest is computed to always be the same.  Constant for `cluster-scoped-resources` please